From 996c05525dbac1256b49964f6cc954b4a65f1469 Mon Sep 17 00:00:00 2001
Message-Id: <patch-1.1-996c05525db-20210906T075527Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Fri, 27 Aug 2021 10:28:41 +0200
Subject: [PATCH] Makefile: remove the check_bindir script
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This script was added in f28ac70f48 (Move all dashed-form commands to
libexecdir, 2007-11-28) when commands such as "git-add" lived in the
bin directory, instead of the git exec directory.

This notice helped someone incorrectly installing version v1.6.0 and
later into a directory built for a pre-v1.6.0 git version.

We're now long past the point where anyone who'd be helped by this
warning is likely to be doing that, so let's just remove this check
and warning to simplify the Makefile.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 Makefile     |  3 +--
 check_bindir | 13 -------------
 2 files changed, 1 insertion(+), 15 deletions(-)
 delete mode 100755 check_bindir

diff --git a/Makefile b/Makefile
index 429c276058d..60c995696c1 100644
--- a/Makefile
+++ b/Makefile
@@ -3099,8 +3099,7 @@ endif
 		  ln "$$execdir/git-remote-http$X" "$$execdir/$$p" 2>/dev/null || \
 		  ln -s "git-remote-http$X" "$$execdir/$$p" 2>/dev/null || \
 		  cp "$$execdir/git-remote-http$X" "$$execdir/$$p" || exit; } \
-	done && \
-	./check_bindir "z$$bindir" "z$$execdir" "$$bindir/git-add$X"
+	done
 
 .PHONY: install-gitweb install-doc install-man install-man-perl install-html install-info install-pdf
 .PHONY: quick-install-doc quick-install-man quick-install-html
diff --git a/check_bindir b/check_bindir
deleted file mode 100755
index 623eadcbb7d..00000000000
--- a/check_bindir
+++ /dev/null
@@ -1,13 +0,0 @@
-#!/bin/sh
-bindir="$1"
-gitexecdir="$2"
-gitcmd="$3"
-if test "$bindir" != "$gitexecdir" && test -x "$gitcmd"
-then
-	echo
-	echo "!! You have installed git-* commands to new gitexecdir."
-	echo "!! Old version git-* commands still remain in bindir."
-	echo "!! Mixing two versions of Git will lead to problems."
-	echo "!! Please remove old version commands in bindir now."
-	echo
-fi
-- 
2.33.0.821.gfd4106eadbd

